### DISCUSSION TITLE
failed to make (Paste Clipboard) script works for multiple selected nodes in (1.11.1-pre28)

### DISCUSSION THREAD

**Discussion by Bricklam** [freeplane/freeplane#1079] [Help]
I tried to execute this script from the script collection page on multiple nodes (to paste a copied branch to multiple nodes at the same time), but it failed:
https://docs.freeplane.org/scripting/Scripts_collection.html#paste-clipboard

here are some of my attempts: (all failed, sometimes: paste the copied tree multiple times to the first node, instead of pasting
one copy to each selected node)

(1)
`// @ExecutionModes({ON_SELECTED_NODE, ON_SELECTED_NODE_RECURSIVELY})

menuUtils.executeMenuItems(['PasteAction',])
`

(2)
`
def toSelect1 = c.selecteds.collect{ 
menuUtils.executeMenuItems(['PasteAction',])
 }.collect()
`

Can anybody help me get it right?
thanks

**Comment by EdoFro** [comment]
```groovy
def mySelectedNodes = [] + c.selecteds

mySelectedNodes.each{n ->
    c.select(n)
    //sleep(1000)
    menuUtils.executeMenuItems(['PasteAction',])
}
```

hope it helps

**Reply by dpolivaev** [reply]
Yes, it is the way to go because PasteAction is always applied to the "main" selected node even if more than one node is selected

**Reply by Bricklam** [reply]
Yes. this script works successfully. exactly what I needed. thanks
